src/libraries/Common/tests/System/Net/Security/Kerberos/KerberosExecutor.cs

```diff
@@ -0,0 +1,143 @@
+
+        _options = new ListenerOptions
+        {
+            Configuration = krb5Config,
+            DefaultRealm = realm,
+            RealmLocator = realm => new FakeRealmService(realm, krb5Config, _principalService),
+            Log = logger,
+            IsDebug = true,
+        };
+
+        _kdcListener = new FakeKdcServer(_options);
+        _realm = realm;
+        _servicePrincipals = new List<FakeKerberosPrincipal>();
+    }
+
+    public void Dispose()
+    {
+        _invokeHandle?.Dispose();
+        _kdcListener.Stop();
+        File.Delete(_krb5Path);
+        File.Delete(_keytabPath);
+    }
+
+    public void AddService(string name, string password = DefaultAdminPassword)
+    {
+        var principal = new FakeKerberosPrincipal(PrincipalType.Service, name, _realm, Encoding.Unicode.GetBytes(password));
+        _principalService.Add(name, principal);
+        _servicePrincipals.Add(principal);
+    }
+ 
+    public void AddUser(string name, string password = DefaultUserPassword)
+    {
+        var principal = new FakeKerberosPrincipal(PrincipalType.User, name, _realm, Encoding.Unicode.GetBytes(password));
+        _principalService.Add(name, principal);
+        _principalService.Add($"{name}@{_realm}", principal);
+    }
+
+    public async Task Invoke(Action method)
+    {
+        await PrepareInvoke();
+        _invokeHandle = RemoteExecutor.Invoke(method);
+    }
+
+    public async Task Invoke(Func<Task> method)
+    {
+        await PrepareInvoke();
+        _invokeHandle = RemoteExecutor.Invoke(method);
+    }
+
+    private async Task PrepareInvoke()
+    {
+        // Start the KDC server
+        var endpoint = await _kdcListener.Start();
+
+        // Generate krb5.conf
+        _krb5Path = Path.GetTempFileName();
+        File.WriteAllText(_krb5Path,
+            OperatingSystem.IsLinux() ?
+            $"[realms]\n{_options.DefaultRealm} = {{\n  master_kdc = {endpoint}\n  kdc = {endpoint}\n}}\n" :
+            $"[realms]\n{_options.DefaultRealm} = {{\n  kdc = tcp/{endpoint}\n}}\n");
+
+        // Generate keytab file
+        _keytabPath = Path.GetTempFileName();
+        var keyTable = new KeyTable();
+
+        var etypes = _options.Configuration.Defaults.DefaultTgsEncTypes;
+        //byte[] passwordBytes = FakeKerberosPrincipal.FakePassword;
+
+        foreach (var servicePrincipal in _servicePrincipals)
+        {
+            foreach (var etype in etypes.Where(CryptoService.SupportsEType))
+            {
+                var kerbKey = servicePrincipal.RetrieveLongTermCredential(etype);
+                keyTable.Entries.Add(new KeyEntry(kerbKey));
+            }
+        }
+
+        using (var fs = new FileStream(_keytabPath, FileMode.Create))
+        using (var writer = new BinaryWriter(fs))
+        {
+            keyTable.Write(writer);
+            writer.Flush();
+        }
+
+        // Set environment variables for GSSAPI
+        Environment.SetEnvironmentVariable("KRB5_CONFIG", _krb5Path);
```

Comment:
does this actually work? Last time (long time ago) I check this did not change the actual ENV e.g. was invisible to native bits.

Reply:
Yes. That's why the test run inside `RemoteExecutor`.